massa-execution-worker/src/execution.rs

```diff
@@ -1507,17 +1508,37 @@ impl ExecutionState {
 
         // return the execution output
         let execution_output = context_guard!(self).settle_slot(None);
-        let exact_cost = req.max_gas.saturating_sub(exec_response.remaining_gas);
+        let gas_used_until_last_subexecution = (context_guard!(self))
+            .gas_remaining_before_subexecution
+            .unwrap_or_default(); // 0 if no subexecutions
+        let exact_exec_cost = req.max_gas.saturating_sub(exec_response.remaining_gas);
+        // compute a gas cost, estimating the gas of the last SC call to be max_instance_cost
+        let corrected_cost = req
+            .max_gas
+            .saturating_sub(gas_used_until_last_subexecution)
```

Comment:
to me it feels like it should be:

```
corrected_cost = gas_used_until_last_subexecution + max_instance_cost
```

what do you think ?

Reply:
I think there might be a confusion between "remaining" and "used" gas until last subexecution

Reply:
I forgot to rename.
Should be clearer now.
The case where `gas_remaining_before_subexecution` is None should also be fixed. It yields 0 so the `exact_exec_cost` will be selected by the max()